ENH 2020 in review (#33)
conda-forge/blog: posts/2020-12-22-year-in-review.rst

   :format: html

2020 in Review
==============

As 2020 winds down, the Core team thought it'd be fun to review some of the big accomplishments our community has made this year.

Strong Growth
-------------

The ``conda-forge`` community has grown immensely this year. Here are a few statistics that measure our growth.

- The community has added 3751 new, unique ``conda`` packages this year, along with a corresponding number of new feedstocks.
- For the majority of 2020, the ``conda-forge`` channel on ``anaconda.org`` exceeded 100 million downloads per month.
- In July of 2020, the ``conda-forge`` channel passed 2 billion total, all-time downloads.
- We've grown our active community, adding seven new members to the ``conda-forge`` Core team and at least two members to
  the ``staged-recipes`` team.
- We now have over 2,500 recipe maintainers in the ``conda-forge`` GitHub organization.

Big New Features
----------------

We've shipped many big updates to our core infrastructure this year. These updates include the move to ``R`` 4.0, the new
CentOS 7 ``sysroot`` for our compilers, completing the ``Python`` 3.8 migration, and the move of our compiler versions to ``GCC`` 9

> and ``clang`` 11. Finally, we've added our first completely cross-compiled platform, ``osx-arm64``, targeting Apple Silicon.

Reviewer: - [x] PyPy
- [x] standalone windows
- [x] miniforge

Reviewer: - [x] Removed python 2 support

.. post:: 2020-12-22
   :tags: conda-forge
   :author: conda-forge/core

.. role:: raw-html(raw)
   :format: html

2020 in Review
==============

As 2020 winds down, the Core team thought it'd be fun to review some of the big accomplishments our community has made this year.

Strong Growth
-------------

The ``conda-forge`` community has grown immensely this year. Here are some numbers to help give you an idea of the

> scale of our growth

Author: ```suggestion
scale of our growth.
```

> - The community has added 3751 new, unique ``conda`` packages this year, along with a corresponding number of new feedstocks.

Author: ```suggestion
- The community has added 3,751 new, unique ``conda`` packages this year, along with a corresponding number of new feedstocks.
```

- For the majority of 2020, the ``conda-forge`` channel on ``anaconda.org`` exceeded 100 million downloads per month.
- In July of 2020, the ``conda-forge`` channel passed 2 billion total, all-time downloads.

> - We've grown our active community, adding seven new members to the ``conda-forge`` Core team and at least two members to

Author: ```suggestion
- We've grown our developer community, adding seven new members to the ``conda-forge`` Core team and at least two members to
```

> - We've grown our developer community, adding seven new members to the ``conda-forge`` Core team and at least two members to

Author: ```suggestion
- We've grown our core developer community, adding seven new members to the ``conda-forge`` Core team and at least two members to
```

- ``R`` **4.0 migration**: This migration was the first one to use our ``automerge`` infrastructure at scale. With it, we
  completed a complete rebuild/upgrade of the ``R`` ecosystem in about a week.
- ``Python`` **updates**: We deprecated ``Python`` 2.7, completed the ``Python`` 3.8 migration, and got about 75% of the way through the
  ``Python`` 3.9 migration.
- **compiler upgrades**: We upgraded our compiler infrastructure to ``GCC`` 9 and ``clang`` 11.
- **CentOS 7 and CentOS 6 EOL**: We shipped an option to enable our compilers to use the CentOS 7 ``sysroot`` in preparation
  for the CentOS 6 EOL. We hope to complete the move to CentOS 7 early next year.
- **miniforge**: We built our own standalone, ``miniconda``-like installers. These support a broad range of platforms, including
  ``osx-arm64`` and ``linux-aarch64``.
- **standalone Windows stack**: We fully decoupled our Windows recipes from the ``defaults`` channel by rebuilding the ``msys2``
  recipes.
- **Apple silicon support**: We added support for Apple silicon with our ``osx-arm64`` platform. This platform is our first
  one to use a fully cross-compiled infrastructure.

> 

Reviewer: ```suggestion
- **CUDA support**: We added support for building CUDA packages on windows and added CUDA 11.0 support.
```

> - **CUDA support**: We added support for building CUDA packages on windows and added CUDA 11.0 support.

Author: ```suggestion
- **CUDA support**: We added support for building CUDA packages on windows and added CUDA 11.0 support.

```
